JavaScriptCode.js

```diff
@@ -0,0 +1,368 @@
+            if(ordernumber >= 1){
+                return parseInt(ordernumber);
+                str = false;
+            }
+            else{
+                console.log(`The order number should be higher than 1.`);
+            }
+        }
+        else{
+            console.log(`Please enter a correct order number where order number should be an integer number higher than 1.`);
+        }
+    }
+}
+
+let myordernumber = checkordernumber();
+console.log(`Hey the order number is : ` + myordernumber);
+const largeboxobject = {size:'Large',value:20,price:1.80};
+const mediumboxobject = {size:'Medium',value:10,price:1.00};
+const smallboxobject = {size:'Small',value:5,price:0.6};
+const bagprice = 5.50;
+const currency = '$';
+let orderId = 1;
+
+/////////////////////////////////////////////////////
+// Start of part A
+/////////////////////////////////////////////////////
+let boxesarrayfun = (par) => {
+    let boxesarray = [];
+    boxesarray.push({size:largeboxobject.size,numbers_of_Bags: largeboxobject.value * parseInt(par / largeboxobject.value), price_of_Boxes:largeboxobject.price * parseInt(par / largeboxobject.value),quantity_of_Size : parseInt(par / largeboxobject.value)});
+    par = par % largeboxobject.value;
+    boxesarray.push({size:mediumboxobject.size,numbers_of_Bags: mediumboxobject.value * parseInt(par / mediumboxobject.value), price_of_Boxes:mediumboxobject.price * parseInt(par / mediumboxobject.value),quantity_of_Size : parseInt(par / mediumboxobject.value)});
+    par = par % mediumboxobject.value;
+    if(par > smallboxobject.value) {
+        boxesarray.push({size:smallboxobject.size,numbers_of_Bags: par, price_of_Boxes:smallboxobject.price * 2,quantity_of_Size : 2});
+    }
+    else if (par > 0){
+        boxesarray.push({size:smallboxobject.size,numbers_of_Bags:  par, price_of_Boxes:smallboxobject.price * 1,quantity_of_Size : 1});
+    }
+    else{
+        boxesarray.push({size:smallboxobject.size,numbers_of_Bags:  par, price_of_Boxes:smallboxobject.price * par,quantity_of_Size : par});
+    }
+    return boxesarray;
+}
+
+let order = (par) => {
+    let bagp = bagprice;
+    let boxesarray = [];
+    let totalnumberofbags, totalcoffeeprice;
+    let totalbagsprice, totalorderprice;
+    let finalobject = {};
+    finalobject["orderID"] = orderId;
+    boxesarray = boxesarrayfun(par);
+    totalnumberofbags = boxesarray.reduce((accumulator, current) => accumulator + (current.numbers_of_Bags), 0);
+    finalobject["totalNumber"] = totalnumberofbags;
+    totalcoffeeprice = totalnumberofbags * bagp;
+    finalobject["totalCoffePrice"] = totalcoffeeprice;
+    totalbagsprice = boxesarray.reduce((accumulator, current) => (accumulator + current.price_of_Boxes ), 0);
+    finalobject["totalBagsPrice"] =  totalbagsprice;
+    totalorderprice = totalcoffeeprice + totalbagsprice;
+    finalobject["totalOrderPrice"] =  totalorderprice;
+    orderId++;
+    let finalarray = finalobject;
```

Comment:
why do we need this?

Reply:
if you mean the order ID we need it in the delete because in my program we can have same order number, means it is not false for example a person to order 300 bags and anther person ordered also 300 bags, so in the delete part when a person want to cancel an order we should be deleting on the order id and not on the order number 300.

Reply:
I meant why do we need to put finalarray = finalobject? and then return final array? can't we return finalobject directly?

Reply:
yes true sorry was having some usage and addition on finalobject and those was removed and should remove the finalarray.

Reply:
done